src/app/archive/lib/processor.ml

```diff
@@ -1358,22 +1391,58 @@ module Zkapp_global_slot_bounds = struct
 end
 
 module Timing_info = struct
-  type t =
-    { account_identifier_id : int
-    ; initial_minimum_balance : string
-    ; cliff_time : int64
-    ; cliff_amount : string
-    ; vesting_period : int64
-    ; vesting_increment : string
-    }
-  [@@deriving hlist, fields]
+  module T = struct
+    type t =
+      { account_identifier_id : int
+      ; initial_minimum_balance : string
+      ; cliff_time : int64
+      ; cliff_amount : string
+      ; vesting_period : int64
+      ; vesting_increment : string
+      }
+    [@@deriving hlist, fields, compare, sexp, hash]
+  end
+
+  include T
 
   let typ =
     Mina_caqti.Type_spec.custom_type ~to_hlist ~of_hlist
       Caqti_type.[ int; string; int64; string; int64; string ]
 
   let table_name = "timing_info"
 
+  let t_to_id = Hashtbl.create (module T)
+
+  let loaded = ref false
+
+  let load (module Conn : CONNECTION) id =
+    Conn.find
+      (Caqti_request.find Caqti_type.int typ
+         (Mina_caqti.select_cols_from_id ~table_name ~cols:Fields.names) )
+      id
+
+  let load_opt (module Conn : CONNECTION) id =
```

Comment:
Why is this function needed?

Reply:
I didn't add it

Reply:
It's used in quite a few tool applications, I didn't review very closely the use-sites but at least one of them didn't need all the values and it felt wasteful to load the whole cache for them.

I agree, these free-floating `load` are unusual.